api/src/main/java/com/github/retrooper/packetevents/protocol/recipe/data/ShapedRecipeData.java

```diff
@@ -21,18 +21,21 @@
 import com.github.retrooper.packetevents.protocol.item.ItemStack;
 import com.github.retrooper.packetevents.protocol.recipe.Ingredient;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 public class ShapedRecipeData implements RecipeData {
     private final int width;
     private final int height;
     private final @NotNull String group;
+    private CraftingBookCategory category;
     private final @NotNull Ingredient[] ingredients;
     private final ItemStack result;
 
-    public ShapedRecipeData(final int width, final int height, final @NotNull String group, final @NotNull Ingredient[] ingredients, final ItemStack result) {
+    public ShapedRecipeData(final int width, final int height, final @NotNull String group, final @NotNull CraftingBookCategory category, final @NotNull Ingredient[] ingredients, final ItemStack result) {
```

Comment:
Is the CraftingBookCategory version-specific? If yes, please make a second constructor